opentelemetry-sdk/src/metrics/data/mod.rs

```diff
@@ -19,6 +19,16 @@ pub struct ResourceMetrics {
     pub scope_metrics: Vec<ScopeMetrics>,
 }
 
+impl ResourceMetrics {
+    /// Returns the total number of metrics across all `ScopeMetrics`.
+    pub fn metrics_count(&self) -> usize {
```

Comment:
lets not expose any new public API. if a component needs it, they can do it on their own for now. While refactoring the collect/export part of metrics, we can discuss what APIs it'd need to expose publically.

Reply:
Fixed this by using the functionality directly where required.